src/main/java/players/mcts/PISingleTreeNode.java

```diff
@@ -0,0 +1,149 @@
+package players.mcts;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import players.PlayerConstants;
+
+import java.util.*;
+
+import static players.PlayerConstants.*;
+
+public class PISingleTreeNode extends SingleTreeNode {
+    SingleTreeNode[] roots;
```

Comment:
A better name for this would be ForestNode (or something that indicates it constructs a number of different trees, and then aggregates results across them)